### DISCUSSION TITLE
How to prevent all pages from refreshing when routing changes, and only change nested pages?

### DISCUSSION THREAD

**Discussion by FatNerdPeng** [nuxt/nuxt#27607] [Questions]
Hi, I have a question. This is my demo structure:
```
components
    Aside.vue
    Top.vue
pages
    index.vue
    hello.vue
    node.vue
app.vue
```
app.vue
```
<template>
  <div>
    <NuxtLayout>
      <NuxtPage />
    </NuxtLayout>
  </div>
</template>
```
index.vue
```
<template>
  <div class="main">
    <div class="top">
      <Top />
    </div>
    <div class="aside">
      <Aside />
    </div>
    <div class="content">
      <NuxtLink to="/hello"> to hello </NuxtLink>
      &nbsp;
      <NuxtLink to="/node">to node</NuxtLink>
      <NuxtPage />
    </div>
  </div>
</template>
```
Aside.vue
```
<script setup lang="ts">
console.log('---aside')
</script>

<template>
  <div>this is aside</div>
</template>
```
Top.vue
```
<script setup lang="ts">
console.log('---top')
</script>

<template>
  <div>this is top</div>
</template>
```
hello.vue
```
<template>
  <div>hello world</div>
</template>
```
node.vue
```
<template>
  <div>hello nodejs</div>
</template>
```
The effect of the `index. vue` now is as follows:
![X2JW 8F3BOI4X_ZQ W_`FKH](https://github.com/nuxt/nuxt/assets/33616855/bc3159a7-b56a-4df3-be8a-bb2e54de9ddd)

However,when I click `to hello` or `to node`, it will lose the top and side areas, directly covering the entire window like this:
![S5WU)_ SNA(QIW_ DRP0(M](https://github.com/nuxt/nuxt/assets/33616855/f8628100-f184-4b1f-b216-aa19a7accaf7)

This is my question. When `<NuxtLink>` points to internal links, isn't it just replacing the content of `<NuxtPage>`?

By the way, I know this requirement can be achieved by writing`<Top>`and`<Aside>`into the `layout`, and then using the `layout` on each `page`. However, there is a problem with this, which is that when switching pages, the page will refresh, causing the JavaScript in the layout to re execute. I don't want this to happen, I just need a partial refresh. I think`<NuxtLink>`can solve this problem because it is a substitute for`<RouterLink>`. 
So where did I make a mistake in writing,thanks!

**Comment by AloisSeckar** [comment]
You have `<NuxtPage>` component defined on top level in your `/app.vue`. So while `<NuxtLink>` is "just replacing the content of `<NuxtPage>`", it effectively means re-rendering everything.

Take the code from `/pages/index.vue` and move it up to `/app.vue`. Then it should work as you intend.

**Reply by FatNerdPeng** [reply]
You're right,I tried according to what you said, but there is a problem with it. For example, my structure may look like this:
```
layouts
  login.vue
  user.vue
  main.vue
pages
  --login
  ----index.vue
  --main
  ----hello.vue
  ----node.vue
  ----index.vue
  --user
  ----front.vue
  ----backend.vue
  ----full.vue
  ...
```
Each directory under `pages` corresponds to a module with different `layout`. I hope that when switching between pages under each layout, only nested pages can be switched. If it is written in `app. vue`, then I can only use one layout or implement different layouts through complex code, which is not what I hope for.

**Reply by maxdzin** [reply]
In `app.vue` specify it like:
```vue
<template>
  <div>
    <NuxtLayout>
      <NuxtPage/>
    </NuxtLayout>
  </div>
</template>
```

Make sure that for each layout (`main`, `login`, `user`) you have a `<slot />` in the template block - there your pages will be rendered.
For example, for `/layouts/main.vue`:
```vue
<template>
  <div class="main">
    <div class="top">
      <Top />
    </div>
    <div class="aside">
      <Aside />
    </div>
    <div class="content">
      <NuxtLink to="/hello"> to hello </NuxtLink>
      &nbsp;
      <NuxtLink to="/node">to node</NuxtLink>

      <!-- This is where your pages will be rendered -->
      <slot />
    </div>
  </div>
</template>
```

Then, you need to specify the layout for every page.
For example, in `/pages/main/hello.vue`
```vue
<script setup lang="ts">
  definePageMeta({
    layout: 'main',
  })
</script>

<template>
  <!-- Hello page content here -->
</template>
```

I would also suggest you take a look at the documentation regarding this:
https://nuxt.com/docs/guide/directory-structure/layouts
https://nuxt.com/docs/guide/directory-structure/app

**Reply by FatNerdPeng** [reply]
Thank you for providing me with a great idea. I think it can help me!

**Reply by maxdzin** [reply]
I'm glad to help

**Comment by maxdzin** [comment]
You need to use one layout for that, like `main`. And then for each of those pages specify:
```ts
  definePageMeta({
    layout: 'main',
  })
```
